packages/integration-sdk-http-client/src/util.ts

```diff
@@ -18,3 +18,44 @@ export function isNonRetryableError(statusCode: number): boolean {
   }
   return false;
 }
+
+export function isRetryableStatusCode(statusCode: number): boolean {
+  if (
+    statusCode === 429 ||
+    statusCode === 408 ||
+    (statusCode >= 500 && statusCode !== 501)
+  ) {
+    return true;
+  }
+  return false;
+}
+
+export async function handleRateLimitError(headers, rateLimitConfig) {
```

Comment:
Not sure about the interface here. I think maybe there should be a unified interface for the sub error handler functions for composability.